vms/proposervm/vm.go

```diff
@@ -392,6 +394,10 @@ func (vm *VM) repair(ctx context.Context) error {
 
 			// innerVM index is incomplete. Wait for completion and retry
 			select {
+			case <-ctx.Done():
+				// "repair" is called with uncancelable context, so this won't be selected
+				// "Done" is here for the code completeness
+				return
```

Comment:
This feels weird to me. I feel like it would generally be reasonable to do something like:
```golang
ctx, cancel := context.WithTimeout(...)
repair(ctx)
cancel()
```
But that doesn't work here. We don't do this, but it feels kind of weird to add an additional invariant to the `repair` caller when it isn't really needed.

Reply:
> weird to add an additional invariant to the repair caller when it isn't really needed.

Fair :)

Let's revisit in the future.